### PR TITLE
Add inline rule example, fix typos

### PR DESCRIPTION
**Pull request by adamkvitek**
# Thanks for improving Semgrep Docs 😀

### Please ensure

- [x] A subject matter expert (SME) reviews the content
- [x] A technical writer reviews the content or PR
- [x] This change has no security implications or else you have pinged the security team
- [x] Redirects are added if the PR changes page URLs

### CODE REVIEW

**Comment by minusworld** [comment]
It make make more sense to put the inline rule example on the overview page, but otherwise lgtm!

**Review by minusworld — APPROVED** [review]

**Review by s-santillan — APPROVED** [review]
lgtm!

File: docs/experiments/join-mode/recursive-joins.md [hunk 1]
```diff
@@ -212,6 +212,55 @@ rule:spring-sql-injection: SQLi
 ran 0 rules on 0 files: 1 findings
 ```
 
+## Recursive join with an inline rule example
```
Comment (minusworld) [inline docs/experiments/join-mode/recursive-joins.md:215]: This is just a regular inline rule, no recursion here :)

"Inline rule example" is perfect for this

Comment (adamkvitek) [reply docs/experiments/join-mode/recursive-joins.md:215]: Thanks... is this even a correct article for this example than actually? 😅 (Title is Recursive joins)

**Pull request merged**